server.js

```diff
@@ -0,0 +1,54 @@
+import express from "express";
+import process  from "os";
+
+const app = express();
+app.use(express.json());
+
+const events =[];
+
+function isValidUTCDate(dateString){
+  const date = new Date(dateString);
+   if(date.toISOString() === dateString){
+    return true;
+   }
+
+  return false;
+}
```

Comment:
```suggestion
function isValidUTCDate(dateString){
 ```suggestion

  const iso8601Regex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d+)?Z$/;

  if (!iso8601Regex.test(dateString))  return false;
 return date.toISOString() === dateString

```

1.  Adicionei uma validação com regex para checar o formato da data. Assim, se a data não for válida, a API enviará a mensagem de erro que você criou, em vez de lançar um erro do Date. Faz uma teste e ve o que acha.
2. Simplifiquei o retorno para  return date.toISOString() === dateString>
    - Código fica mais simples e claro e você evita um bloco if desnecessário.
}
```